asreview/webapp/authentication/oauth_handler.py

```diff
@@ -0,0 +1,154 @@
+import requests
+
+
+class OAuthHandler:
+    def __init__(self, configs=None):
+        if not (bool(configs) and isinstance(configs, dict)):
+            raise f"OAuthHandler needs a configuration dictionary."
+
+        # check if all necessary config parameters are there
+        services = {}
+        for provider, config in configs.items():
+            # get required parameters
+            authorization_url = config.get("AUTHORIZATION_URL", False)
+            token_url = config.get("TOKEN_URL", False)
+            client_id = config.get("CLIENT_ID", False)
+            secret = config.get("CLIENT_SECRET", False)
+            scope = config.get("SCOPE", "")
+            if not (
+                all(
+                    [
+                        bool(authorization_url),
+                        bool(token_url),
+                        bool(client_id),
+                        bool(secret),
+                    ]
+                )
+            ):
+                raise f"OAuthHandler has insufficient data for f{provider}"
+            else:
+                # rebuild config
+                services[provider.lower()] = {
+                    "authorization_url": authorization_url,
+                    "token_url": token_url,
+                    "client_id": client_id,
+                    "secret": secret,
+                    "scope": scope,
+                }
+        # set the config dictionary
+        self.services = services
+
+    def front_end_params(self):
+        """prepare a service dictionary for front-end: remove secrets"""
```

Comment:
There should be clear documentation how to use `OAuthHandler` without showing secrets anywhere.